### DISCUSSION TITLE
why is there a tokio::time::Instant ?

### DISCUSSION THREAD

**Discussion by barkanido** [tokio-rs/tokio#3470] [Q&A] [answered]
What is the justification and motivation for introducing `tokio::time::Instant`? 
Is it maybe to better support `no_std` environments?

**Comment by Darksonn** [comment, marked as answer]
It is because when you enable the `test-util` feature, it is possible to [pause time][1], which can be very useful in tests.

[1]: https://docs.rs/tokio/1/tokio/time/fn.pause.html

**Reply by barkanido** [reply]
so, internally, Tokio has to use the `tokio::time` wrapper instead of `std::time`?
In other words, in order to pause time in tests for code that I write using Tokio, I need to use `tokio::time` instead?

**Reply by Darksonn** [reply]
Yes, if you want to use that feature, you should be using Tokio's `Instant`. Tokio cannot override the std `Instant`, and the std `Instant` will give the real time even if you have paused it with `tokio::time::pause`.

Of course, you can always convert between the std and Tokio `Instant` types via the `from_std` and `into_std` methods on the Tokio `Instant`.
